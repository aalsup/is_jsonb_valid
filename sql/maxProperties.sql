-- maxProperties validation
-- shorter is valid
SELECT is_jsonb_valid('{"maxProperties":2}', '{"foo":1}');
-- exact length is valid
SELECT is_jsonb_valid('{"maxProperties":2}', '{"foo":1,"bar":2}');
-- too long is invalid
SELECT is_jsonb_valid('{"maxProperties":2}', '{"foo":1,"bar":2,"baz":3}');
-- ignores arrays
SELECT is_jsonb_valid('{"maxProperties":2}', '[1,2,3]');
-- ignores strings
SELECT is_jsonb_valid('{"maxProperties":2}', '"foobar"');
-- ignores other non-objects
SELECT is_jsonb_valid('{"maxProperties":2}', '12');