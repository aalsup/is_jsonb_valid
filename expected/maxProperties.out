-- maxProperties validation
-- shorter is valid
SELECT is_jsonb_valid('{"maxProperties":2}', '{"foo":1}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- exact length is valid
SELECT is_jsonb_valid('{"maxProperties":2}', '{"foo":1,"bar":2}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- too long is invalid
SELECT is_jsonb_valid('{"maxProperties":2}', '{"foo":1,"bar":2,"baz":3}');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores arrays
SELECT is_jsonb_valid('{"maxProperties":2}', '[1,2,3]');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores strings
SELECT is_jsonb_valid('{"maxProperties":2}', '"foobar"');
 is_jsonb_valid 
----------------
 f
(1 row)

-- ignores other non-objects
SELECT is_jsonb_valid('{"maxProperties":2}', '12');
 is_jsonb_valid 
----------------
 f
(1 row)
